src/App.tsx

```diff
@@ -1,44 +1,99 @@
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User>();
+  const [selectedPost, setSelectedPost] = useState<Post>();
+  const [loader, setLoader] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
+  const [createNewComment, setCreateNewComment] = useState<boolean>(false);
+
+  useEffect(() => {
+    getUsersData()
+      .then(setUsers)
+      .catch(() => setErrorMessage(true));
+  }, []);
+
+  useEffect(() => {
+    if (selectedUser) {
+      setLoader(true);
+      setErrorMessage(false);
+
+      getPostsData(selectedUser.id)
+        .then(setUserPosts)
+        .catch(() => setErrorMessage(true))
+        .finally(() => setLoader(false));
+    }
+  }, [selectedUser]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                  setSelectedPost={setSelectedPost}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                { loader && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {errorMessage && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {
+                  (selectedUser && userPosts.length === 0
+                    && !errorMessage && !loader) && (
+                    <div
+                      className="notification is-warning"
+                      data-cy="NoPostsYet"
+                    >
+                      No posts yet
+                    </div>
+                  )
+                }
 
-                <PostsList />
+                {selectedUser && userPosts.length > 0 && (
```

Comment:
you can convert to boolean like this:
```suggestion
                {selectedUser && !!userPosts.length && (
```